app/shared/components/BigButton.svelte

```diff
@@ -0,0 +1,64 @@
+<script type="ts" context="module">
+
+export function generateIcon(preferredJourneyMode:string){
+  if(preferredJourneyMode === "fastest") return "north"
+  if(preferredJourneyMode === "least-changes") return "switch_access_shortcut_add"
+  if(preferredJourneyMode === "least-walking") return "directions_walk"
+
+  return ""
+}
+
+</script>
+
+<script type="ts">
+
+  import { createEventDispatcher } from "svelte";
+  import { PreferredJourneyMode } from "~/types";
+
+  export let icon = ""
+  $: hasIcon = icon !== ""
+  export let label = "Default Label";
+
+</script>
+<gridLayout class="button-style">
+    <stackLayout orientation="horizontal" on:tap>
+      <label class={hasIcon
+        ? "icon button-withIcon"
+        : "button-withoutIcon" } text="{icon}"/>
```

Comment:
You can actually render code dynamically

```suggestion
      {#if hasIcon}
      <label class="icon" text="{icon}"/>
      {/endif}
```

Reply:
klappt nicht

Reply:
Oh sorry endif ist falsch nur /if